client/chat/channel-banner-reducer.ts

```diff
@@ -0,0 +1,56 @@
+import { Immutable } from 'immer'
+import { ChannelBannerId, ChannelBannerJson } from '../../common/chat-channels/channel-banners'
+import { immerKeyedReducer } from '../reducers/keyed-reducer'
+
+export interface ChannelBannerState {
+  /** An array of channel banners that are available to us. */
+  availableChannelBanners: ChannelBannerId[]
+  /** An array of default channel banners. */
+  defaultChannelBanners: ChannelBannerId[]
+  /** A map of channel banner ID -> channel banner info (used in channel info card, etc.) */
```

Comment:
nit: Comments that just repeat the type info are probably not that useful, we only used to make these comments because we didn't have types